### DISCUSSION TITLE
Json Table format

### DISCUSSION THREAD

**Discussion by MecMax** [Zefau/ioBroker.jarvis#2200] [Q&A]
Hallo liebe Gemeinde!
Ich versuche verzweifelt, bestimmte Werte meiner JsonTable in Jarvis farbig darzustellen. Ich habe folgende Version im Einsatz:
v3.1.2 (Pro-Version)
Ich habe eine Tabelle, in welcher ich den Online-Status meiner Zigbee-Geräte abfrage und in Jarvis visualisiere. Klapt auch wunderbar. Nun möchte ich die Werte der Spalte Status farbig darstellen (Offline=rot; Online=grün). Leider scheitere ich daran.
Bei meinen Tests ist es mir gelungen, den Text zu ändern (Spalten JSON-Format):
[{"field": "Device", "label": "Device"  }, { "label": "Status", "align":"center", "format":"val => val == 'Online' ? 'ONLINE' : val" }]
Ich würde nun gerne nicht nur den Text ändern, sondern auch die Farbe (Online:grün;Offline:rot).
{"Online":{"color":"#00ff00"},"default":{"color":"#ff0000","fontWeight":"bold"}}
Ist das beim Format-Parameter überhaupt möglich? Oder muß ich anders ansetzen. Hat hier jemand eine Idee?
MfG
MecMax

**Comment by mcuiobroker** [comment]
https://mcuiobroker.gitbook.io/jarvis-infos/jarvis-v3/besonderheiten-v3/module/jsontable/v3.x.x/parameter-style

**Comment by MecMax** [comment]
hallo @mcuiobroker !
Ich habe den Parameter aus der Anleitung daptiert und verwendet. Die Darstellung zeigt leider beim Style keine Reaktion. Auf den Parameter Format reagiert die Darstellung. Hier mein Style:
[  {    "field": "Device",    "label": "Device"  },  {    "field": "Status",    "label": "Status",  "style": "width:20px;background-color: #ff0000;font-size:20px"  }]
MfG MecMax

**Reply by mcuiobroker** [reply]
Ja, scheint ein BUG zu sein.
Aufgenommen https://github.com/Zefau/ioBroker.jarvis/issues/2202

**Comment by mcuiobroker** [comment]
@hgschiller ? Hast du es gelöscht ?

**Comment by hgschiller** [comment]
Ich würde auch gerne den Style ändern, abhängig von val ! Aber hab es bisher nicht hinbekommen.
Wir auch ganz oben beschrieben möchte ich den Style so ändern, das (Online:grün;Offline:rot). Allerdings nicht die Schrift sondern der Hintergrund.

**Reply by mcuiobroker** [reply]
Dann musst du nicht "color" setzen, sondern "background-color".
![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/504c089c-b6c9-4d4c-ac1e-8e3e86ec3421)
PRO?

**Reply by hgschiller** [reply]
Ich habe die Pro , aber wo ist die Abhängigkeit zum Inhalt ? mit Style eine Spalte setzen ist klar, aber wie geht das bei dem Fall der oben beschrieben ist, Sprich im Feld selber steht Online oder Offline und ich möchte den Style bei Online anders haben als bei Offline ?
Aber schon mal danke für die super schnelle Antwort !

**Reply by mcuiobroker** [reply]
Wodurch wird Tabelle erstellt? 
Selbst erstellt,  dann kann man mit HTML-Befehlen arbeiten, ansonsten mit dem callback. (row)=>{...}

**Reply by hgschiller** [reply]
Ok ! Aber wie müsste so eine Zeile mit Callback aussehen ? 
Beispiel:
{"field":"Status","style":"(row) => val === 'Online' ? 'background-color: #00FF00' : 'background-color: #FF0000' "}
Sorry, wenn ich mich ein wenig unbeholfen hier anstelle.

**Reply by hgschiller** [reply]
Ach ja Tabelle wird von mir selber als json in ein Objekt abgelegt.
[
  {
    "Server": "Name des Rechners",
    "Status": "Online oder Offline"
  }
...
]

**Reply by mcuiobroker** [reply]
Hab ich selbst auch noch nicht gemacht.
In der Doku sind console.log Beispiele wie man sich rantasten kann. Ich muss mal später schauen.
(row, props)=>{if(props.col.value == 'Online')...}

**Reply by hgschiller** [reply]
erst mal merci !

**Reply by mcuiobroker** [reply]
Wenn du sie selbst erstellst dann ist einfach.
```
Statt "online"
"<div style=\"background-color:red;\">online</div>'
```

**Reply by hgschiller** [reply]
Danke noch mal für dein Feedback, so bin ich auf die Lösung gekommen:
"style": "(rowIndex, row, props) => (row.Status == 'Online' ? 'background-color: #00FF00' : 'background-color: #FF0000')"
Danke noch mal und schönen Sonntag noch.

**Reply by mcuiobroker** [reply]
Mit row.Status sollte es nicht funktionieren, es muss props.Status sein.
```
"style": "(rowIndex, row, props) => (props.Status == 'Online' ? 'background-color: #00FF00' : 'background-color: #FF0000')"
```
Es liegt an den Variablen, die man mit gibt.
Hat man (rowindex,row,props) -> row.Status funktioniert.
nimmt man nur (sowie ich) (row,props) -> funktionert es nicht mit row.Status, sondern nur mit props.Status.
